src/laser/MultiLaser.cpp

```diff
@@ -1017,24 +1017,26 @@ MultiLaser::InitLaserSlice (const amrex::Geometry& geom, const int islice, const
                 bx,
                 [=] AMREX_GPU_DEVICE(int i, int j, int k)
                 {
-                    amrex::Real z = plo[2] + (islice+0.5_rt)*dx_arr[2] - zfoc;
-                    const amrex::Real x = (i+0.5_rt)*dx_arr[0]+plo[0]-x0;
-                    const amrex::Real y = (j+0.5_rt)*dx_arr[1]+plo[1]-y0;
-
+                    amrex::Real z = plo[2] + (islice+0.5_rt)*dx_arr[2] - z0;
+                    amrex::Real x = (i+0.5_rt)*dx_arr[0]+plo[0]-x0;
+                    amrex::Real y = (j+0.5_rt)*dx_arr[1]+plo[1]-y0;
+                    const amrex::Real yp=std::cos(init_angle_yz)*y+std::sin(init_angle_yz)*z;
+                    const amrex::Real zp=-std::sin(init_angle_yz)*y+std::cos(init_angle_yz)*z;
+                    z=zp;
+                    y=yp;
                     // For first laser, setval to 0.
                     if (ilaser == 0) {
                         arr(i, j, k, comp ) = 0._rt;
                         arr(i, j, k, comp + 1 ) = 0._rt;
                     }
-
                     // Compute envelope for time step 0
-                    Complex diffract_factor = 1._rt + I * z * 2._rt/( k0 * w0 * w0 );
+                    Complex diffract_factor = 1._rt + I * (z-zfoc*std::cos(init_angle_yz)) * 2._rt/( k0 * w0 * w0 );
```

Comment:
Could you check if this correction is needed? e.g. by running with zero and non-zero (but small) angle, and checking where the laser is focused?